testsuite/MDAnalysisTests/converters/test_rdkit.py

```diff
@@ -529,37 +581,128 @@ def test_reassign_props_after_reaction(self, rdmol, product, name):
+        "N[C@@H](CS)C(=O)O",  # C
+        "CSCC[C@H](N)C(=O)O",  # M
+        "NC(=O)C[C@H](N)C(=O)O",  # N
+        "NC(=O)CC[C@H](N)C(=O)O",  # Q
+        # nucleic acids
+        "Nc1ncnc2c1ncn2[C@H]1C[C@H](OP(=O)(O)O)[C@@H](COP(=O)(O)O)O1",  # A
+        "Cc1cn([C@H]2C[C@H](OP(=O)(O)O)[C@@H](COP(=O)(O)O)O2)c(=O)[nH]c1=O",  # T
+        "O=c1ccn([C@H]2C[C@H](OP(=O)(O)O)[C@@H](COP(=O)(O)O)O2)c(=O)[nH]1",  # U
+        "Nc1ccn([C@H]2C[C@H](OP(=O)(O)O)[C@@H](COP(=O)(O)O)O2)c(=O)n1",  # C
+        "Nc1nc2c(ncn2[C@H]2C[C@H](OP(=O)(O)O)[C@@H](COP(=O)(O)O)O2)c(=O)[nH]1",  # G
     ])
-    def test_order_independant(self, smi_in):
+    def test_order_independant(self, smi):
         # generate mol with hydrogens but without bond orders
-        ref = Chem.MolFromSmiles(smi_in)
-        template = Chem.AddHs(ref)
-        for atom in template.GetAtoms():
-            atom.SetIsAromatic(False)
-            atom.SetFormalCharge(0)
-        for bond in template.GetBonds():
-            bond.SetIsAromatic(False)
-            bond.SetBondType(Chem.BondType.SINGLE)
-
-        # go through each possible starting atom
-        for a in template.GetAtoms():
-            smi = Chem.MolToSmiles(template, rootedAtAtom=a.GetIdx())
-            m = Chem.MolFromSmiles(smi, sanitize=False)
-            for atom in m.GetAtoms():
-                atom.SetFormalCharge(0)
-                atom.SetNoImplicit(True)
-            m.UpdatePropertyCache(strict=False)
-            _infer_bo_and_charges(m)
-            m = _standardize_patterns(m)
-            Chem.SanitizeMol(m)
+        ref = Chem.MolFromSmiles(smi)
+        stripped_mol = self.add_Hs_remove_bo_and_charges(ref)
+        # enumerate mols with different starting atoms
+        for m in self.enumerate_reordered_mol(stripped_mol):
+            m = self.assign_bond_orders_and_charges(m)
             m = Chem.RemoveHs(m)
-            assert m.HasSubstructMatch(ref) and ref.HasSubstructMatch(
-                m), (f"(input) {Chem.MolToSmiles(ref)} != "
-                     f"{Chem.MolToSmiles(m)} (output) root atom {a.GetIdx()}")
+            self.assert_isomorphic_resonance_structure(m, ref)
+
+    @pytest.mark.xfail(reason="Not currently tackled by the RDKitConverter")
+    @pytest.mark.parametrize("smi", [
+        "C-[N+]#N",
+        "C-N=[N+]=[N-]",
+        "C-[O+]=C",
+        "C-[N+]#[C-]",
+    ])
+    def test_order_independant_issue_3339(self, smi):
+        self.test_order_independant(smi)
 
     def test_warn_conjugated_max_iter(self):
         smi = "[C-]C=CC=CC=CC=CC=CC=C[C-]"
         mol = Chem.MolFromSmiles(smi)
         with pytest.warns(UserWarning,
                           match="reasonable number of iterations"):
             _rebuild_conjugated_bonds(mol, 2)
+
+    @pytest.mark.parametrize("smi", [
+        "[Li+]", "[Na+]", "[K+]", "[Rb+]", "[Ag+]", "[Cs+]",
+        "[Mg+2]", "[Ca+2]", "[Cu+2]", "[Zn+2]", "[Sr+2]", "[Ba+2]",
+        "[Al+3]", "[Fe+2]",
```

Comment:
What happens for `"[Fe+3]"` or `"[Cu+1]"`?

Reply:
If nothing else, document known failure cases or make choices explicit in the docs. E.g., it's fine to say "Any iron ion will be guessed to be Fe3+" etc. We just need to be clear about what happens.

Reply:
> What happens for `"[Fe+3]"` or `"[Cu+1]"`?

Those will always be Fe2+ and Cu2+ with the code in this PR. We could solve this by looking at naming conventions for these ions but that's for another PR I guess. I'll add it to issue #3339 though